File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int device_num = nums.size()/2;
    
    sort(nums.begin(), nums.end());
    
    nums.erase(unique(nums.begin(), nums.end()), nums.end());
    
    if(device_num <= nums.size()) return device_num;
    else if(device_num > nums.size()) return nums.size();
    
    
    return answer;
}

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.java
import java.util.Arrays;
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
		int answer = 0;
		int cnt = nums.length/2;
		Arrays.sort(nums);
		Set<Integer> set = new HashSet<Integer>();
		
		for(int a : nums) {
			set.add(a);
		}
		
		answer = set.size();
		
        if(answer > cnt) {
			return cnt;
		}else {
			return answer;
			
		}
}
}